File: python_leaoai/python_leaoai/models/Factory/ProjectBlock.py
# models/project_block.py

from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship
from datetime import datetime
from database import Base
from .project import Project
from .project_phase import ProjectPhase
from .project_item import ProjectItem
from .project_group import ProjectGroup
from .project_measure_item import ProjectMeasureItem
from .project_image import ProjectImage

class ProjectBlock(Base):
    __tablename__ = "project_block"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)

    project_id: Mapped[int] = mapped_column(ForeignKey("project.id"))
    project: Mapped["Project"] = relationship("Project")

    project_phase_id: Mapped[int] = mapped_column(ForeignKey("project_phase.id"))
    project_phase: Mapped["ProjectPhase"] = relationship("ProjectPhase")

    project_item_id: Mapped[int] = mapped_column(ForeignKey("project_item.id"))
    project_item: Mapped["ProjectItem"] = relationship("ProjectItem")

    project_group_id: Mapped[int] = mapped_column(ForeignKey("project_group.id"))
    project_group: Mapped["ProjectGroup"] = relationship("ProjectGroup")

    usuario_id: Mapped[int | None] = mapped_column(Integer, nullable=True)

    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    position: Mapped[int | None] = mapped_column(Integer, nullable=True)

    code: Mapped[str] = mapped_column(String, nullable=False)
    name: Mapped[str | None] = mapped_column(String, nullable=True)
    type: Mapped[str | None] = mapped_column(String, nullable=True)

    min_images_amount: Mapped[int] = mapped_column(Integer, default=0)
    max_images_amount: Mapped[int] = mapped_column(Integer, default=0)
    images_label: Mapped[str | None] = mapped_column(String, nullable=True)
    observations_enabled: Mapped[bool] = mapped_column(Boolean, default=False)
    is_completed: Mapped[bool] = mapped_column(Boolean, default=False)
    instructions: Mapped[str | None] = mapped_column(String, nullable=True)
    box_type: Mapped[str | None] = mapped_column(String, nullable=True)
    card_board_type: Mapped[str | None] = mapped_column(String, nullable=True)
    closure_type: Mapped[str | None] = mapped_column(String, nullable=True)
    action_text: Mapped[str | None] = mapped_column(String, nullable=True)
    is_measurable_items: Mapped[bool] = mapped_column(Boolean, default=False)
    version: Mapped[str | None] = mapped_column(String, nullable=True)
    observation: Mapped[str | None] = mapped_column(String, nullable=True)

    box_type_active: Mapped[bool] = mapped_column(Boolean, default=False)
    card_board_type_active: Mapped[bool] = mapped_column(Boolean, default=False)
    closure_type_active: Mapped[bool] = mapped_column(Boolean, default=False)
    version_active: Mapped[bool] = mapped_column(Boolean, default=False)
    block_for_conclude: Mapped[bool] = mapped_column(Boolean, default=False)

    ativo: Mapped[bool] = mapped_column(Boolean, default=True)
    excluido: Mapped[bool] = mapped_column(Boolean, default=False)

    # Relacionamentos
    project_measure_items: Mapped[list["ProjectMeasureItem"]] = relationship(
        "ProjectMeasureItem", back_populates="project_block"
    )
    project_images: Mapped[list["ProjectImage"]] = relationship(
        "ProjectImage", back_populates="project_block"
    )

File: Framework.Data/Models/Factory/ProjectBlock.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Framework.Data.Models.Factory
{
	// PROJECT BLOCK
	public class ProjectBlock
	{
		[Key]
		public int Id { get; set; }

		[ForeignKey("Project")]
		public int ProjectId { get; set; }
		public virtual Project Project { get; set; }

		[ForeignKey("ProjectPhase")]
		public int ProjectPhaseId { get; set; }
		public virtual ProjectPhase ProjectPhase { get; set; }

		[ForeignKey("ProjectItem")]
		public int ProjectItemId { get; set; }
		public virtual ProjectItem ProjectItem { get; set; }

		[ForeignKey("ProjectGroup")]
		public int ProjectGroupId { get; set; }
		public virtual ProjectGroup ProjectGroup { get; set; }

		// Se necessário, referenciar a tabela Usuario
		public int? UsuarioId { get; set; }

		public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
		public DateTime? UpdatedAt { get; set; }
		public int? Position { get; set; }

		[Required]
		public string Code { get; set; }  // Gerado como Guid.ToString()

        public string Name { get; set; }  // Gerado como Guid.ToString()

        public string Type { get; set; }
		public int MinImagesAmount { get; set; } = 0;
		public int MaxImagesAmount { get; set; } = 0;
		public string ImagesLabel { get; set; }
		public bool ObservationsEnabled { get; set; } = false;
		public bool IsCompleted { get; set; } = false;
		public string Instructions { get; set; }
		public string BoxType { get; set; }
		public string CardBoardType { get; set; }
		public string ClosureType { get; set; }
		public string ActionText { get; set; }
		public bool IsMeasurableIitems { get; set; } = false;
		public string Version { get; set; }
		public string Observation { get; set; }

		// Campos adicionais conforme solicitado
		public bool BoxTypeActive { get; set; } = false;
		public bool CardBoardTypeActive { get; set; } = false;
		public bool ClosureTypeActive { get; set; } = false;
		public bool VersionActive { get; set; } = false;
		public bool BlockForConclude { get; set; } = false;

		public bool Ativo { get; set; } = true;
		public bool Excluido { get; set; } = false;

		// Navegação
		public virtual ICollection<ProjectMeasureItem> ProjectMeasureItems { get; set; } = new List<ProjectMeasureItem>();
		public virtual ICollection<ProjectImage> ProjectImages { get; set; } = new List<ProjectImage>();
	}
}
